src/dsp_tools/commands/ingest_xmlupload/bulk_ingest_client.py

```diff
@@ -82,3 +84,36 @@ def upload_file(
         print(f"{datetime.now()}: {msg}")
         logger.info(msg)
         return None
+
+    def kick_off_ingest(self) -> None:
+        """Start the ingest process on the server."""
+        url = f"{self.dsp_ingest_url}/projects/{self.shortcode}/bulk-ingest"
+        res = self.session.post(url, timeout=5)
+        if res.status_code == STATUS_CONFLICT:
+            msg = f"Ingest process on the server {self.dsp_ingest_url} is already running. Wait until it completes..."
+            print(msg)
+            logger.info(msg)
+        if res.json().get("id") != self.shortcode:
+            raise UserError("Failed to kick off the ingest process.")
+        print(f"Kicked off the ingest process on the server {self.dsp_ingest_url}. Wait until it completes...")
+        logger.info(f"Kicked off the ingest process on the server {self.dsp_ingest_url}. Wait until it completes...")
+
+    def retrieve_mapping(self) -> str | None:
+        """Try to retrieve the mapping CSV from the server."""
+        url = f"{self.dsp_ingest_url}/projects/{self.shortcode}/bulk-ingest/mapping.csv"
+        res = self.session.get(url, timeout=5)
+        if res.status_code == STATUS_CONFLICT:
+            print("Ingest process is still running. Wait until it completes...")
+            logger.info("Ingest process is still running. Wait until it completes...")
+            return None
+        elif not res.ok or not res.text.startswith("original,derivative"):
+            msg = (
+                "Dubious error while polling for the mapping CSV. "
```

Comment:
Is it even meaningful to do this check that it starts with `original,derivative`? can it ever start with anything else?

Also, I don't like the word dubious... maybe "unexpected"? however, I think 404 might be something we get at times?

I think I would split this test up into two distinct cases:
1. `if not res.ok:` some error message indicating that the server did not return a mapping
2. `if not res.text:` some error indicating that the server returned an empty mapping

Reply:
> Is it even meaningful to do this check that it starts with original,derivative? can it ever start with anything else?
It should never start with anything else, but I think it is meaningful to check even for events that should not happen.

> Also, I don't like the word dubious
I adapted the wording.

> I think I would split this test up into two distinct cases
I only see 3 major cases:
- 409 Conflict: Do nothing, wait for next poll.
- non-200, or content of response is not a mapping: inform the user, wait for next poll. (Perhaps, the server recovers)
- 200: return the mapping

Reply:
in that case I would only check these 3 cases. For the "non-200" I would build in some kind of maximum number of attempts because if the Server doesn't recover, it should not wait forever.

Reply:
Done with https://github.com/dasch-swiss/dsp-tools/commit/3285fbfe175e3fa3d3b1fd25c40347ba763fbed2